plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -13,38 +13,56 @@
  */
 package io.trino.plugin.password.ldap;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.ImmutableSet;
-import io.trino.plugin.password.Credential;
+import com.google.common.io.Closer;
 import io.trino.spi.security.AccessDeniedException;
 import io.trino.spi.security.BasicPrincipal;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
-import javax.naming.NamingException;
-
-import java.util.HashSet;
-import java.util.Optional;
-import java.util.Set;
-
+import static io.trino.plugin.password.ldap.LdapUtil.randomSuffix;
+import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 
 public class TestLdapAuthenticator
 {
-    private static final String BASE_DN = "base-dn";
-    private static final String PATTERN_PREFIX = "pattern::";
+    private final Closer closer = Closer.create();
+
+    private TestingOpenLdapServer openLdapServer;
+    private LdapAuthenticatorClient client;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        openLdapServer = new TestingOpenLdapServer();
+        closer.register(openLdapServer);
+        openLdapServer.start();
+
+        client = new JdkLdapAuthenticatorClient(new LdapConfig()
+                .setLdapUrl(openLdapServer.getLdapUrl()));
+    }
+
+    @AfterClass
+    public void close()
+            throws Exception
+    {
+        closer.close();
+    }
 
     @Test
     public void testSingleBindPattern()
+            throws Exception
     {
-        TestLdapAuthenticatorClient client = new TestLdapAuthenticatorClient();
-        client.addCredentials("alice@example.com", "alice-pass");
+        String organizationDistinguishedName = openLdapServer.createOrganization("organization_" + randomSuffix());
```

Comment:
It would be nice to be able drop that organization after the test. 

Maybe:

```
try(Organization organization = openLdapServer.createOrganization("organization_" + randomSuffix())) {
   String organizationDistinguishedName = organization.getName();
```
?   

Same for user and other objects we create in shared ldap service.

Reply:
Addressed it.